pkg/sql/sem/builtins/generator_builtins.go

```diff
@@ -1261,13 +1261,8 @@ func makeJSONPopulateImpl(gen tree.GeneratorWithExprsFactory, info string) tree.
 		// the default values of each field will be NULL.
 		// The second argument can also be null, in which case the first argument
 		// is returned as-is.
-		Types: tree.ArgTypes{{"base", types.Any}, {"from_json", types.Jsonb}},
-		ReturnType: func(args []tree.TypedExpr) *types.T {
-			if len(args) != 2 {
-				return tree.UnknownReturnType
-			}
-			return args[0].ResolvedType()
-		},
+		Types:              tree.ArgTypes{{"base", types.AnyTuple}, {"from_json", types.Jsonb}},
```

Comment:
The reason I set this to be `types.Any` is to match the argument type in Postgres itself, which is any, and not tuple. I'm not sure if this matters, but that's why I did it:

```
postgres=# select proargtypes[0]::regtype from pg_proc where proname = 'jsonb_populate_record';
 proargtypes
-------------
 anyelement
(1 row)
```

Reply:
hm, so looks like PG must do some other validation on the input args
```
rafiss@127:postgres> select * from json_populate_record(null::int, '{"a":1,"b":2}')
first argument of json_populate_record must be a row type
```

if you try this in crdb:
```
root@:26257/defaultdb> select * from json_populate_record(null::record, '{"a":1,"b":2}');
ERROR: anonymous records cannot be used with json_populate_record
```

which seems like a misleading message

Reply:
interesting, postgres caches records, wonder if we should do something similar:
https://github.com/postgres/postgres/blob/4548c76738b368a11a5dad052f9653a349eeb52c/src/backend/utils/adt/jsonfuncs.c#L3815-L3825

anyway yeah it's checked internally i think? the NULL is checked here
https://github.com/postgres/postgres/blob/4548c76738b368a11a5dad052f9653a349eeb52c/src/backend/utils/adt/jsonfuncs.c#L3416-L3422

though i can't find where it returns the signature error if it is a not-null not-tuple. maybe we should just change the `tree.MustBeDTuple` check here: https://github.com/cockroachdb/cockroach/blob/50396a923da22a9685fee5c3c4248e88c921ab8e/pkg/sql/sem/builtins/generator_builtins.go#L1321

Reply:
It looks like PG checks the not-null not-tuple case in the same place as above. there's also this fun message that PG has, which probably doesn't apply to CRDB's type system:
```
rafiss@127:postgres> select * from json_populate_recordset(null, '{"a":1,"b":2}')
could not determine polymorphic type because input has type unknown
```